File: pubnub-java/src/main/java/com/pubnub/api/models/consumer/pubsub/PNMessageResult.java
package com.pubnub.api.models.consumer.pubsub;

import com.google.gson.JsonElement;

import com.pubnub.api.PubNubError;
import lombok.Getter;
import lombok.ToString;
import org.jetbrains.annotations.Nullable;

@Getter
@ToString(callSuper = true)
public class PNMessageResult extends MessageResult {

    @Nullable
    private final PubNubError error;

    public PNMessageResult(BasePubSubResult basePubSubResult, JsonElement message) {
        this(basePubSubResult, message, null);
    }

    public PNMessageResult(BasePubSubResult basePubSubResult, JsonElement message, @Nullable PubNubError error) {
        super(basePubSubResult, message);
        this.error = error;
    }
}

File: pubnub-kotlin-core/core-api/src/main/kotlin/com/pubnub/api/models/consumer/pubsub/PNMessageResult.kt
package com.pubnub.api.models.consumer.pubsub

import com.google.gson.JsonElement
import com.pubnub.api.PubNubError

/**
 * Wrapper around an actual message.
 */
data class PNMessageResult(
    private val basePubSubResult: PubSubResult,
    override val message: JsonElement,
    val error: PubNubError? = null
) : MessageResult, PubSubResult by basePubSubResult
